frontend/src/metabase/lib/pulse.ts

```diff
@@ -106,32 +113,35 @@ export function pulseIsValid(pulse, channelSpecs) {
   );
 }
 
-export function dashboardPulseIsValid(pulse, channelSpecs) {
+export function dashboardPulseIsValid(
+  pulse: Pulse,
+  channelSpecs: ChannelSpecs,
+) {
   return pulseChannelsAreValid(pulse, channelSpecs);
 }
 
-export function emailIsEnabled(pulse) {
+export function emailIsEnabled(pulse: Pulse) {
   return (
     pulse.channels.filter(c => c.channel_type === "email" && c.enabled).length >
     0
   );
 }
 
-export function cleanPulse(pulse, channelSpecs) {
+export function cleanPulse(pulse: Pulse, channelSpecs: any) {
   return {
     ...pulse,
     channels: cleanPulseChannels(pulse.channels, channelSpecs),
     parameters: cleanPulseParameters(getPulseParameters(pulse)),
   };
 }
 
-function cleanPulseChannels(channels, channelSpecs) {
+function cleanPulseChannels(channels: Channel[], channelSpecs: any) {
   return channels.filter(c =>
-    channelIsValid(c, channelSpecs && channelSpecs[c.channel_type]),
+    channelIsValid(c, channelSpecs?.[c.channel_type]),
```

Comment:
nit: since you've already changed `c` to `channel` on line 91 above. Should you do the same for other places? Namely here and line 125

Reply:
Definitely @WiNloSt. Updated 👍